lib/dialect/src/ConstraintsAnalysis.cpp

```diff
@@ -0,0 +1,1028 @@
+				return this->ranges;
+			}
+
+			//TODO: understand what this meet does
+			// In theory it should be the opposite of a join
+			mlir::ChangeResult meet(
+					const mlir::dataflow::AbstractDenseLattice& val) override
+			{
+				llvm::outs()<<"meet\n";
+				//return mlir::ChangeResult::NoChange;
+				// Do a funny thing and let the meet perform the join
+				return this->join(val);
+			}
+
+			mlir::ChangeResult join(
+					const mlir::dataflow::AbstractDenseLattice& val) override
+			{
+				//Need to cast it
+				const auto& other = *static_cast<const ConstraintsLattice*>(&val);
+				
+				//Keep it to check if there is no change
+				auto copy = *this;
+
+				llvm::outs()<<"Did i join?\n";
+				this->print(llvm::outs());
+				other.print(llvm::outs());
+
+				//Insert the other in this
+				for(auto other_range : other.ranges){
+					
+					// We have to insert the other range
+					// If we find it then we have to join, else we have to insert it
+					this->insertOrJoin(other_range.first, other_range.second);
+				}
+
+				//
+				if (*this==copy){
+					llvm::outs()<<"no\n";
+					this->print(llvm::outs());
+					return mlir::ChangeResult::NoChange;
+				}
+				this->print(llvm::outs());
+				return mlir::ChangeResult::Change;
+			}
+
+			// Operation to copy two lattices
+			// NB: this is a deep copy
+			mlir::ChangeResult copy(const ConstraintsLattice& other)
+			{
+				llvm::outs()<<"Did i copy?\n";
+				llvm::outs()<<"mine : ";this->print(llvm::outs());
+				llvm::outs()<<"other: ";other.print(llvm::outs());
+
+				if (*this==other){
+					llvm::outs()<<"no\n";
+					return mlir::ChangeResult::NoChange;
+				}
+				this->ranges=other.ranges;
+				return mlir::ChangeResult::Change;
+			}
+
+			mlir::ChangeResult copy(const mlir::dataflow::AbstractDenseLattice& val){
+				return this->copy(*static_cast<const ConstraintsLattice*>(&val));
+			}
+
+			void print(raw_ostream& os) const override
+			{
+				os<<"Ranges:\n";
+				if(not this->ranges.empty()){
+					for(auto range: this->ranges){
+						os << "SSA: "<< range.first.first <<" Branch: "<< range.first.second << 
+						" RANGE: (" << (range.second.hasMin() ? std::to_string(range.second.getMin()):"?") 
+						<< ", " << (range.second.hasMax() ? std::to_string(range.second.getMax()):"?") << ")\n";
+					}
+				}
+				else {
+					os<<"Empty\n";
+				}
+			}
+
+			bool operator==(const ConstraintsLattice& other) const
+			{
+				return this->ranges == other.ranges;
+			}
+
+			mlir::ChangeResult setCurrentBranch(mlir::RegionBranchOpInterface* branchOp){
+				// The yield has this format
+				// %12 = rlc.constant true !rlc.bool
+    			// rlc.yield %12 : !rlc.bool
+				// So I need to retrieve "true" in this case
+
+				// Here we get the rlc.constant true
+				auto value=
+					branchOp->getOperation() // gets the rlc.yield
+					->getOperand(0); // get the first operand of the yield (%12:!rlc.bool)
```

Comment:
prefer
``` 
mlir::cast<mlir::rlc::Yield>(branchOp->getOperation()).getArguments()[0];
```

the difference is that if the syntax of yield changes somehow and ends up with extra operands at index 0, this version i have written will still extract the right operand, while looking inside operation internals is not safe.

Reply:
Ok! Done!